TEST Build from main branch with code updates to icd11 component goal and problematic exclusion script

twhetzel opened the pull request:
Resolves #ISSUE(s).
<!--- "Resolves #ISSUE" will automatically close #ISSUE when the PR is merged. However, if this PR won't 100% address 
the issue and you don't want it to auto-close, you can write one of the following instead of "Resolves"
- Partially addresses
- Addresses sub-task (n) in
-->

## Overview
<!--- A few sentences describing changes / what problem is solved. Should describe the expected result of the changes. 
If applicable, point out the main change that solved the problem, e.g. fixed bug in method xyz. -->
This PR:
Includes code updates to:
- `mondo-ingest.Makefile` to remove `--trim false` in the goal `$(COMPONENTSDIR)/icd11foundation.owl`
- update to SPARQL query used for label look-up for curies identified as problematic exclusions to check for labels using first `rdfs:label` and then `skos:prefLabel`. Currently, ICD10CM and icd11.foundation both use `skos:prefLabel` in the files in TMPDIR

This PR also includes data updates based on a full build of the mondo-ingest pipeline run as: `sh run.sh make build-mondo-ingest -B 2>&1 | tee logs.txt`

NOTE: Keep in mind this is the first data build since icd11.foundation was included in "mondo.sssom.tsv" following the inclusion of xrefs to icd11.foundation for the August Mondo release and subsequent [updates in the mondo repo](https://github.com/monarch-initiative/mondo/pull/8054) to make sure the icd11.foundation xrefs were included in the "mondo.sssom.tsv" file.

PROPOSAL: Given this very special situation where this is Week 2 (Data) of the Mondo Release Cycle where lexical alignments are reviewed and generally new builds are done _and_ that this hotfix code change is needed to fix the build on `main`, I suggest that if we approve this PR and it's ok with Sabrina that both the code and data files be merged into `main`. @matentzn and @twhetzel will discuss Wednesday.
- Update: Sabrina is ok if this is merged into `main`.

## Pre-merge checklist
<!--- Docs: A common case for documentation is the addition of new `make` goals. Descriptions should be documented for 
new goals both in the (i) `help` command at the bottom of the `mondo-ingest.Makefile` and (ii) 
`docs/developer/workflows.md`. -->

### Documentation
Was the documentation added/updated under `docs/`?
- [ ] Yes
- [x] No, updates to the docs were not necessary after careful consideration

### QC
Was the full pipeline run before submitting this PR using `sh run.sh make build-mondo-ingest` on this branch (after 
`docker pull obolibrary/odkfull:dev`), and no errors occurred?
- [x] Yes
- [ ] No, there are no functional (code-related) changes to the pipeline in the PR, so no re-run is necessary
   
### New Packages
Were any new *Python packages* added?
- [ ] Yes, [requirements.txt.full](https://github.com/INCATools/ontology-development-kit/blob/master/requirements.txt.full) was updated and an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) submitted
- [x] No

Were any other *non-Python packages* added?
- [ ] Yes, [Dockerfile](https://github.com/INCATools/ontology-development-kit/blob/master/Dockerfile) updated and an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) submitted
- [x] No

### PR Review and Conversations Resolved
Has the PR been sufficiently reviewed by at least 1 team member of the Mondo Technical team and all threads resolved?
- [ ] Yes

joeflack4 commented on src/ontology/reports/icd11foundation_excluded_terms_in_mondo_xrefs.tsv:
## Overall review
The bug fix solved the `RuntimeError` and the `icd11foundation_excluded_terms_in_mondo_xrefs.tsv` and `icd11foundation_excluded_terms_in_mondo_xrefs_summary.tsv` are generated. Good!

I was expecting to see an updated file that concats all sources: `excluded_terms_in_mondo_xrefs.tsv`, but it hasn't been updated. However, I now see that this  file isn't run as part of the pipeline.

I haven't taken a general face check over all the other outputs but I can take a glance tomorrow if needed.

joeflack4 replied:
Generally passes my face check looking at all the files overall.

joeflack4 replied:
@twhetzel Did you get the changes you expected when running the build?

I see [here](https://github.com/monarch-initiative/mondo-ingest/pull/634#discussion_r1715656734) you said:
> The set of problematic excluded terms is now larger (which I believe is expected).

I assume that before, this larger set was simply the addition of these new rows added to `icd11foundation_excluded_terms_in_mondo_xrefs.tsv`, and no other changes, right?

Just wanna make sure that these build results are consistent w/ what you got before running the build.

twhetzel replied:
Yes, we can discuss on the Tech call this Friday if the same change to $(COMPONENTSDIR)/icd11foundation.owl is needed for other sources. No need to add any tickets for this at this time.

joeflack4 reviewed (approved):
LGTM!

twhetzel commented:
Nico and I discussed the change in this PR and we decided to go with the fix included in this PR. We refactored the SPARQL query and as a test I removed the icd11.foundation reports (icd11foundation_excluded_terms_in_mondo_xrefs.tsv and icd11foundation_excluded_terms_in_mondo_xrefs_summary.tsv) and then re-ran that specific goal as: `sh run.sh make reports/icd11foundation_excluded_terms_in_mondo_xrefs.tsv -B` and the reports were re-generated without any changes so another full build is _not_ needed. 

We also decided that in this special case it is ok to merge both this code hotfix and any data changes into `main` since it is Week 2 - Build Week of the Mondo Release Cycle.
